src/main/java/ch/kalunight/zoe/service/leaderboard/RankLeaderboardService.java

```diff
@@ -1,160 +1,170 @@
+  }
+
+  @Override
+  protected void runLeaderboardRefresh(Server server, Guild guild, TextChannel channel, Leaderboard leaderboard,
+      Message message) throws SQLException, RiotApiException {
+
+    Objective objective = Objective.getObjectiveWithId(leaderboard.lead_type);
+    QueueSelected queueSelected = null;
+    if(Objective.SPECIFIC_QUEUE_RANK.equals(objective)) {
+      queueSelected = gson.fromJson(leaderboard.lead_data, QueueSelected.class);
+    }
+
+    List<PlayerRank> playersRank = orderAndGetPlayers(guild, objective, queueSelected);
+
+    List<String> playersName = new ArrayList<>();
+    List<String> dataList = new ArrayList<>();
+
+    for(PlayerRank playerRank : playersRank) {
+      playersName.add(playerRank.getPlayer().getUser().getAsMention());
+      FullTier fullTier = playerRank.getFullTier();
+      if(queueSelected == null) {
+        dataList.add(Ressources.getTierEmote().get(fullTier.getTier()).getUsableEmote() + " " + fullTier.toString(server.serv_language) 
+        + " (" + LanguageManager.getText(server.serv_language, playerRank.getQueue().getNameId()) + ")");
+      }else {
+        dataList.add(Ressources.getTierEmote().get(fullTier.getTier()).getUsableEmote() + " " + fullTier.toString(server.serv_language));
+      }
+    }
+
+    String playerTitle = LanguageManager.getText(server.serv_language, "leaderboardPlayersTitle");
+    String dataName;
+    if(queueSelected == null) {
+      dataName = LanguageManager.getText(server.serv_language, "leaderboardRankTitle");
+    }else {
+      dataName = String.format(LanguageManager.getText(server.serv_language, "leaderboardRankSpecificQueueTitle"), LanguageManager.getText(server.serv_language, queueSelected.getGameQueue().getNameId()));
+    }
+
+    EmbedBuilder builder = buildBaseLeaderboardList(playerTitle, playersName, dataName, dataList);
+    builder.setColor(Color.ORANGE);
+    if(queueSelected == null) {
+      builder.setTitle(LanguageManager.getText(server.serv_language, "leaderboardObjectiveRankAllTitle"));
+      message.editMessage(LanguageManager.getText(server.serv_language, "leaderboardObjectiveRankAllTitle")).queue();
+    }else {
+      builder.setTitle(String.format(LanguageManager.getText(server.serv_language, "leaderboardObjectiveRankSpecificTitle"), 
+          LanguageManager.getText(server.serv_language, queueSelected.getGameQueue().getNameId())));
+      message.editMessage(String.format(LanguageManager.getText(server.serv_language, "leaderboardObjectiveRankSpecificTitle"), 
+          LanguageManager.getText(server.serv_language, queueSelected.getGameQueue().getNameId()))).queue();
+    }
+
+    builder.setFooter(LanguageManager.getText(server.serv_language, "leaderboardRefreshMessage"));
+    message.editMessage(builder.build()).queue();
+
+  }
+
+  private List<PlayerRank> orderAndGetPlayers(Guild guild, Objective objective, QueueSelected queueSelected) throws SQLException, RiotApiException {

```

Comment:
Codacy found an issue: [The method 'orderAndGetPlayers(Guild, Objective, QueueSelected)' has an NPath complexity of 363, current threshold is 200](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=6283923)